hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/DefinitionsService.java

```diff
@@ -37,31 +37,132 @@
     /*
         CRUD interface for Trigger
      */
+
+    /**
+     * Create a new <code>Trigger</code>.  <code>Conditions</code> and <code>Actions</code> are manipulated in separate
+     * calls. The new </code>Trigger</code> will be persisted.  When fully defined a call to
+     * {@link #updateTrigger(Trigger)}
+     * is needed to enable the </code>Trigger</code>.
+     * @param trigger
+     * @throws Exception If the </code>Trigger</code> already exists.
+     */
     void addTrigger(Trigger trigger) throws Exception;
+
+    /**
+     * The <code>Trigger</code> will be removed from the Alerts engine, as needed, and will no longer be persisted.
+     * @param triggerId
+     * @throws Exception
+     */
     void removeTrigger(String triggerId) throws Exception;
-    void updateTrigger(Trigger trigger) throws Exception;
-    Collection<Trigger> getTriggers() throws Exception;
+
+    /**
+     * Update the <code>Trigger</code>. <code>Conditions</code> and <code>Actions</code> are manipulated in separate
+     * calls. The updated </code>Trigger</code> will be persisted.  If enabled the </code>Trigger</code>
+     * will be [re-]inserted into the Alerts engine and any prior dampening will be reset.
+     * @param triggerId
+     * @throws Exception If the </code>Trigger</code> does not exist.
+     */
+    Trigger updateTrigger(Trigger trigger) throws Exception;
+
     Trigger getTrigger(String triggerId) throws Exception;
 
+    Collection<Trigger> getAllTriggers() throws Exception;
+
     /*
-        CRUD interface for Trigger
+        CRUD interface for Dampening
+     */
+
+    Dampening addDampening(Dampening dampening) throws Exception;
+
+    void removeDampening(String dampeningId) throws Exception;
+
+    Dampening updateDampening(Dampening dampening) throws Exception;
+
+    Dampening getDampening(String dampeningId) throws Exception;
```

Comment:
The original Trigger object was actually a heavy object, which we normalized given the original (current) CRUD API.  So this change was supporting that normalized approach.  Actually, Triggers encompass not only dampening but also the conditions.  I am OK with re-designing the model and API such that Trigger comprises its dampening and conditions.  What we need to avoid is something in the middle, which is what we had before this proposed change.

Dampening and Condition will still need to be objects understood by the calling code, we are talking only about their access, and possible their storage.   I think Dampening and Condition should be treated the same way in the API.